2_process/calc_doy_means.py

```diff
@@ -16,17 +16,16 @@ def calc_doy_means(df, site_id, out_file=None):
         doy_means.to_csv(out_file)
     return doy_means
 
-def main(out_dir, in_dir, lake_ids):
+def main(out_file, in_file, lake_id):
+    out_dir = os.path.dirname(out_file)
     if not os.path.exists(out_dir):
 	       os.makedirs(out_dir)
-    for i in lake_ids:
-        pred_csv_file = os.path.join(in_dir,
-                                 f"pgdl_nhdhr_{i}_temperatures.csv")
-        df = read_pred_csv(pred_csv_file)
-        calc_doy_means(df, i, f"2_process/out/doy_{i}.csv")
+    df = read_pred_csv(in_file)
+    calc_doy_means(df, lake_id, f"2_process/out/doy_{lake_id}.csv")
 
 if __name__ == '__main__':
-    out_dir = "2_process/out"
-    in_dir = "1_fetch/out/tmp/"
     lake_ids = ["120020150", "107072210"]
-    main(out_dir, in_dir, lake_ids)
+    for i in lake_ids:
+        out_file = f"2_process/out/doy_{i}.csv"
+        in_file = f"1_fetch/out/tmp/pgdl_nhdhr_{i}_temperatures.csv"
+        main(out_file, in_file, i)
```

Comment:
```suggestion
    for lake_id in lake_ids:
        out_file = f"2_process/out/doy_{lake_id}.csv"
        in_file = f"1_fetch/out/tmp/pgdl_nhdhr_{lake_id}_temperatures.csv"
        main(out_file, in_file, lake_id)
```
The variable name `i` that we were using before was not very informative - I suggest we change it.